### ISSUE TITLE
Guinier model does not allow negative Rg^2^ (Trac #1112)

### ISSUE THREAD

**Issue by butlerpd** [SasView/sasview#1157]
After the Stuhrmann Analysis experiment for the NIST summer school it has been noted that the Gunier model in !SasView is technically not generally correct as it does not allow for a negative Rg^2^.  While this is not seen very often, it is the basis of Stuhrmann analysis and happens essentially with heterogeneous scattering length densities near the contrast point.  In the short term some update to the documentation would be good.  In the longer term we should think about making Rg^2^ the fit parameter.

Migrated from http://trac.sasview.org/ticket/1112
```json
{
    "status": "closed",
    "changetime": "2018-11-14T00:04:04",
    "_ts": "2018-11-14 00:04:04.483199+00:00",
    "description": "After the Stuhrmann Analysis experiment for the NIST summer school it has been noted that the Gunier model in !SasView is technically not generally correct as it does not allow for a negative Rg^2^.  While this is not seen very often, it is the basis of Stuhrmann analysis and happens essentially with heterogeneous scattering length densities near the contrast point.  In the short term some update to the documentation would be good.  In the longer term we should think about making Rg^2^ the fit parameter.",
    "reporter": "butler",
    "cc": "",
    "resolution": "fixed",
    "workpackage": "SasView Bug Fixing",
    "time": "2018-06-25T17:22:50",
    "component": "SasView",
    "summary": "Guinier model does not allow negative Rg^2^",
    "priority": "major",
    "keywords": "",
    "milestone": "SasView 4.2.0",
    "owner": "tcbennun",
    "type": "defect"
}
```

**Comment by butlerpd** [comment]
Trac update at `2018/06/26 01:10:14`: 
* **butler** commented:

oops ... not a 4.2 item clearly.


* **butler** changed milestone from "SasView 4.2.0" to "SasView 4.3.0"

**Comment by RichardHeenan** [comment]
Trac update at `2018/06/26 10:40:06`: **richardh** commented:

Suggest for backwards compatibilty that we allow negative Rg value, and change the exponent so that

Rg_Squared = abs(Rg)*Rg

I will assign this to our student as an intro to sasview.

**Comment by RichardHeenan** [comment]
Trac update at `2018/06/28 15:21:36`: **richardh** commented:

Negative Rg!^2 has been sorted in sasmodels ticket-1112 branch

However when I build the docs the demo plot goes down to 10!^-300 as seems not to pick up the flat background of 0.001, this needs further investigation.

**Comment by butlerpd** [comment]
Trac update at `2018/07/04 17:31:05`: **butler** commented:

Interestingly in versions <4.0 Guinier model did NOT include a background parameter.  Since 4.0.x it does probably happened when sasmodels infrastructure took control of scale and background in order to more easily do mixture models (to prevent problems like Richard mentions of having (P(Q)+BGD) * (S(Q) + BGD) ....

However it does seem that somewhere along the line the translation to the auto generation of the docs does not pass that Background.  This needs to be a new ticket.

What I don't understand however is that for this Guinier even the scale factor seems to be wrong with I(0) being or order 10^-20^ while other models seem to be correct.  Moreover, it seems to be correct in master where the Guinier I(0) is 1 (10^0^).  I cannot see any code change that would cause that so this needs to be checked (perhaps it is a flakiness of my build system?)

Otherwise I think this is ready for a pull request though I would like to add a bit in the documentation about Rg (I like the current addition which is an excellent usage note) as it seems that a lot of our users are unclear on the concept: namely that it it quantifies the distribution from the "center of mass" of some parameter where that parameter in mechanics is mass (which can never be negative and is related to what users like to think of) but in scattering is the scattering length density (which can be negative and is not the same as mass distribution unless the object has a uniform scattering length density).  Further it is only R,,g,,^2^ which is defined with R,,g,,=&radic;R,,g,,^2^, and R,,g,,^2^=&sum; {&rho;,,i,, (r,,i,,-r,,0,,)^2^}
Where the sum is over all points i in the object and r,,0,, is the center of mass of the object.

**Comment by butlerpd** [comment]
Trac update at `2018/07/07 16:04:34`: **butler** commented:

Wow .. nice FAST job! .... except that I lead you astray :-(  sorry about that!  Forgot that R,,g,,^2^ has to be normalized by &sum;&rho;,,i,, ... also it would probably be clearer if it read "SLD center of mass" (i.e. not mass center of mass) in the two places "center of mass" is used.  I believe that "center of mass" is a general term that applies to any intrinsic variable like mass or SLD etc (e.g. the population center of mass?)but it is confusing when trying to distinguish between MASS center of mass and something else center of mass.

I suggest submitting a pull request from this branch. I think it is ready to go with these minor changes

**Comment by butlerpd** [comment]
Trac update at `2018/07/12 01:16:10`: 
* **butler** commented:

OOps ... merged without closing the ticket -- doing so now


* **butler** changed resolution from "" to "fixed"
* **butler** changed status from "assigned" to "closed"

**Comment by butlerpd** [comment]
Trac update at `2018/11/14 00:04:04`: 
* **butler** commented:

was fixed for 4.2 so moving to that milestone now.


* **butler** changed milestone from "SasView 4.3.0" to "SasView 4.2.0"
